### ISSUE TITLE
Cell Context menu event should maintain the cell selection when triggered within the selection

### ISSUE THREAD

**Issue by seancolsen** [centerofci/mathesar#1771]
## Steps to reproduce

1. Select a block of cells.
1. Trigger the context menu on a cell outside the selected block.
1. Observe the cell selection to change, selecting only the cell on which the context menu was triggered. (good)
1. Select a block of cells again.
1. Trigger the context menu on a cell _inside_ the selected block.
1. Expect the cell selection to remain unchanged when the context menu opens. This behavior will be important when/if we add actions within the context menu that operate on multiple cells/rows/columns.
1. Observe the cell selection to change, selecting only the cell on which the context menu was triggered. (bad)

**Comment by monstajoe2002** [comment]
Can you assign me this issue? 
Also, by context menu you mean a custom right-click menu?

**Comment by seancolsen** [comment]
Sure @monstajoe2002, go ahead. 

> you mean a custom right-click menu?

Yes. "Context menu" is the proper name since it can be opened through other means too, e.g. the "Menu" key on some keyboards, and Ctrl+Click on MacOS.

**Comment by monstajoe2002** [comment]
Can I use a third party library to implement it?

**Comment by seancolsen** [comment]
@monstajoe2002, no. We already have a context menu system. No need for any additional libraries. This ticket is just requesting an adjustment to the behavior of the cell selection when opening the context menu. I expect it to be a very small change.

**Comment by rajatvijay** [comment]
@seancolsen I had a question on this issue: 

Since we do not have any action in the context menu as of now that works on multiple cells together, what is that menu supposed to show?

**Comment by seancolsen** [comment]
Now that we have the ability to delete multiple records at once, I'm marking this ticket as ready, and I've updated the description to specify the content of the context menu when opened on multiple cells.

**Comment by zackkrida** [comment]
Marking this as user reported since a user mentioned it to me today.
